compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -93,6 +93,8 @@ object Splicer {
     }
   }
 
+  private val classLoaders = scala.collection.mutable.HashMap.empty[String, URLClassLoader]
+
```

Comment:
It means the cache is going to be shared by all compiler runs?  I'm not sure about the following:

- concurrency
- memory leak
- setting changes between runs

I'm not familiar with these areas, could you explain?

Reply:
I changed the `Splicer` to a class and instantiate one per run to avoid the issues above.

Reply:
It is not to share `s.c.m.HashMap` between threads without synchronization.

I'm thinking about using a timed cache in Scalac to deal with the leak problem (keep the loader around for X seconds after last use).

Reply:
The other issue with caching classloaders is having a good plan for invalidating them when JARs and directories on disk change.

Reply:
Now the cache is only valid during one phase of one run of the compiler. None of these should be an issue.